src/main/java/io/github/robwin/ratelimiter/RateLimiterConfig.java

```diff
@@ -18,46 +18,60 @@
  */
 package io.github.robwin.ratelimiter;
 
-import static java.util.Objects.requireNonNull;
-
 import java.time.Duration;
 
+import static java.util.Objects.requireNonNull;
+
 public class RateLimiterConfig {
     private static final String TIMEOUT_DURATION_MUST_NOT_BE_NULL = "TimeoutDuration must not be null";
     private static final String LIMIT_REFRESH_PERIOD_MUST_NOT_BE_NULL = "LimitRefreshPeriod must not be null";
     private static final Duration ACCEPTABLE_REFRESH_PERIOD = Duration.ofNanos(1L);
 
-    private final Duration timeoutDuration;
-    private final Duration limitRefreshPeriod;
-    private final int limitForPeriod;
+    private final Context context;
```

Comment:
Maybe we can get rid of Context by making all fields private non final? And builder will just set this during construction.